### PR TITLE
Docker_Image.rst: Add configuration for Circle CI

### LINKED ISSUE #379: Add CircleCI to Users/Docker_Image
Users/Docker_Image currently provides configuration for GitLab CI and Travis CI.

Circle CI should also be added.
See https://circleci.com/docs/docker/

### PR DESCRIPTION
**Pull request by vijeth-aradhya**
Add configuration information for Circle CI on use
of the `coala/base` docker image.

Closes #379

### CODE REVIEW

File: Users/Docker_Image.rst [hunk 1]
```diff
@@ -69,3 +69,25 @@ on your code with a ``.travis.yml``, like this:
...
+    machine:
+      services:
+        - docker
+
+    test:
+      override:
+        - docker run -v=$(pwd):/app --workdir=/app coala/base coala --ci
+
+.. note::
+
```
Comment (nemaniarjun) [inline Users/Docker_Image.rst]: Something isnt right here. It does not come under the note.

Comment (Mixih) [inline Users/Docker_Image.rst]: Imo this is fine.... Can you eleborate @nemaniarjun ?

File: Users/Docker_Image.rst [hunk 2]
```diff
@@ -69,3 +69,25 @@ on your code with a ``.travis.yml``, like this:
...
+  For more information about Circle CI configuration, consult the
+  `official documentation <https://circleci.com/docs/>`__.
```
Comment (Mixih) [inline Users/Docker_Image.rst]: Please add this newline at the end of the file

Comment (vijeth-aradhya) [inline Users/Docker_Image.rst]: I have!! This was the first thing I fixed..

Comment (Mixih) [inline Users/Docker_Image.rst]: ah well, github is acting up then... sorry :/

File: Users/Docker_Image.rst (hunk 1 above)
Comment (nemaniarjun) [inline Users/Docker_Image.rst]: Yeah my bad, I thought the `::` after note will format the actual note text :P

File: Users/Docker_Image.rst (hunk 2 above)
Comment (fneu) [inline Users/Docker_Image.rst]: btw. why would SpaceConsistencyBear propose to just delete the last line of the file in this case? :joy:

File: Users/Docker_Image.rst [hunk 3]
```diff
@@ -69,3 +69,25 @@ on your code with a ``.travis.yml``, like this:
...
+      services:
+        - docker
+
+    test:
+      override:
+        - docker run -v=$(pwd):/app --workdir=/app coala/base coala --ci
+
+.. note::
+
+  For more information about Circle CI configuration, consult the
```
Comment (Mixih) [inline Users/Docker_Image.rst]: I think indentation is off here

File: Users/Docker_Image.rst (hunk 2 above)
Comment (aptrishu) [inline Users/Docker_Image.rst]: Ah there is a problem with the unified_diff method()..

File: Users/Docker_Image.rst (hunk 3 above)
Comment (Mixih) [inline Users/Docker_Image.rst]: NVM, github is being stupid

**Comment by Mixih** [comment]
ack f3af46

**Comment by Mixih** [comment]
ack 7f3af46

**Comment by Mixih** [comment]
@rultor merge

**Pull request merged**